plugins/kotlin/injection/src/org/jetbrains/kotlin/idea/injection/KotlinLanguageInjectionContributor.kt

```diff
@@ -167,6 +168,31 @@ class KotlinLanguageInjectionContributor : LanguageInjectionContributor {
             ?: injectInAnnotationCall(place)
             ?: injectWithReceiver(place)
             ?: injectWithVariableUsage(place, originalHost)
+            ?: injectWithMutation(place)
+    }
+
+    private val stringMutationOperators = listOf(KtTokens.EQ, KtTokens.PLUSEQ)
+    private fun injectWithMutation(host: KtElement): InjectionInfo? {
+        val parent = (host.parent as? KtBinaryExpression)?.takeIf { it.operationToken in stringMutationOperators } ?: return null
+        if (parent.right != host) return null
+
+        if (isAnalyzeOff(host.project)) return null
+
+        val property = when (val left = parent.left) {
+            is KtQualifiedExpression -> left.selectorExpression
+            else -> left
+        } ?: return null
+
+        for (reference in property.references) {
+            ProgressManager.checkCanceled()
+            val resolvedTo = reference.resolve()
+            if (resolvedTo is KtProperty) {
+                val annotation = resolvedTo.findAnnotation(FqName(AnnotationUtil.LANGUAGE)) ?: return null
```

Comment:
I want to use `injectWithExplicitCodeInstruction` here, but it works on the strings themselves, not the definition site. Changing them seems too big a change.